pkg/admission/admissiontimeout/timeoutadmission.go

```diff
@@ -0,0 +1,63 @@
+package admissiontimeout
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/apiserver/pkg/admission"
+)
+
+type pluginHandlerWithTimeout struct {
+	name            string
+	admissionPlugin admission.Interface
+	timeout         time.Duration
+}
+
+var _ admission.ValidationInterface = &pluginHandlerWithTimeout{}
+var _ admission.MutationInterface = &pluginHandlerWithTimeout{}
+
+func (p pluginHandlerWithTimeout) Handles(operation admission.Operation) bool {
+	return p.admissionPlugin.Handles(operation)
+}
+
+func (p pluginHandlerWithTimeout) Admit(a admission.Attributes, o admission.ObjectInterfaces) error {
+	mutatingHandler, ok := p.admissionPlugin.(admission.MutationInterface)
+	if !ok {
+		return nil
+	}
+
+	admissionDone := make(chan struct{})
+	var admissionErr error
+	go func() {
+		admissionErr = mutatingHandler.Admit(a, o)
+	}()
+
+	select {
+	case <-admissionDone:
+		return admissionErr
+	case <-time.After(p.timeout):
+		return errors.NewTimeoutError(fmt.Sprintf("admission plugin %q failed to complete mutation in %v", p.name, p.timeout), int(p.timeout.Seconds()+1))
```

Comment:
why +1 ?

Reply:
why the conversion to int? Isn't `"13s"` not good enough?

Reply:
TimeoutError requires a time to retry in.  I want to make sure its never zero, so plus one.  And it's required to be an int. I can reorder if you like